backend/src/plugins/camera/camera_service_impl.h

```diff
@@ -121,6 +123,542 @@ class CameraServiceImpl final : public rpc::camera::CameraService::Service {
+        auto stream_closed_future = stream_closed_promise.get_future();
+
+        bool is_finished = false;
+
+        _camera.subscribe_possible_setting_options(
+            [&writer, &stream_closed_promise, &is_finished](
+                const std::vector<dronecode_sdk::Camera::SettingOptions> setting_options) {
+                rpc::camera::PossibleSettingOptionsResponse rpc_setting_options_response;
+
+                for (const auto setting_option : setting_options) {
+                    auto rpc_setting_option = rpc_setting_options_response.add_setting_options();
+                    translateSettingOptions(setting_option, rpc_setting_option);
+                }
+
+                if (!writer->Write(rpc_setting_options_response) && !is_finished) {
+                    is_finished = true;
+                    stream_closed_promise.set_value();
+                }
+            });
+
+        stream_closed_future.wait();
+
+        return grpc::Status::OK;
+    }
+
+    static void translateSettingOptions(const dronecode_sdk::Camera::SettingOptions setting_options,
+                                        rpc::camera::SettingOptions *rpc_setting_options)
+    {
+        rpc_setting_options->set_setting_id(setting_options.setting_id);
+
+        for (const auto option : setting_options.options) {
+            auto rpc_option = rpc_setting_options->add_options();
+            rpc_option->set_option_id(option.option_id);
+        }
+    }
+
+    static dronecode_sdk::Camera::SettingOptions
+    translateRPCSettingOptions(const rpc::camera::SettingOptions rpc_setting_options)
+    {
+        dronecode_sdk::Camera::SettingOptions setting_options;
+        setting_options.setting_id = rpc_setting_options.setting_id();
+
+        std::vector<dronecode_sdk::Camera::Option> options;
+        for (auto option : rpc_setting_options.options()) {
+            options.push_back(translateRPCOption(option));
+        }
+        setting_options.options = options;
+
+        return setting_options;
+    }
+
+    grpc::Status SetSetting(grpc::ServerContext * /* context */,
+                            const rpc::camera::SetSettingRequest *request,
+                            rpc::camera::SetSettingResponse *response) override
+    {
+        std::promise<void> set_option_called_promise;
+        auto set_option_called_future = set_option_called_promise.get_future();
+
+        if (request == nullptr) {
+            if (response != nullptr) {
+                fillResponseWithResult(response, dronecode_sdk::Camera::Result::WRONG_ARGUMENT);
+            }
+        } else {
+            const std::string setting_id = request->setting().setting_id();
+            dronecode_sdk::Camera::Option option;
+            option.option_id = request->setting().option().option_id();
+
+            _camera.set_option_async(
+                [this, response, &set_option_called_promise](
+                    dronecode_sdk::Camera::Result camera_result) {
```

Comment:
Don't you need to ignore results such as `IN_PROGRESS` and maybe others before setting the response here?

Reply:
Oh, I don't know ^^. So `set_option_async()` may return `IN_PROGRESS`?

Reply:
Yes, I think that can potentially happen. Just ignore that result.

Reply:
I suppose so yes. Not all have that result but for calibration it's used because the process can take a while.

Reply:
Are you sure I need to do that? It seems to me that `IN_PROGRESS` is some kind of error (like a specialization of `BUSY`). It happens if you call `SetSetting` again before the previous call finished, right?

Reply:
You receive `IN_PROGRESS` when the camera is working on the command. So it's not an error but rather an information.

Reply:
I suggest you add it and then merge it.

Reply:
I see. Will do!